File: campushub-frontend/src/services/marksService.js
import api from "../api/axiosConfig";

export const getAllMarks = async () => {
    const response = await api.get("/marks");
    return response.data.data;
};

export const saveMarks = async (marks) => {
    const response = await api.post("/marks/bulk", marks);
    return response.data.data;
};

export const getMarksByCourse = async (courseId) => {
    const response = await api.get(`/marks/course/${courseId}`);
    return response.data.data;
};

File: src/main/java/com/preeti/campushub/service/MarksService.java
package com.preeti.campushub.service;

import java.util.List;

import com.preeti.campushub.dto.marks.MarksRequest;
import com.preeti.campushub.dto.marks.MarksResponse;

public interface MarksService {

    MarksResponse createMarks(MarksRequest request);

    List<MarksResponse> getAllMarks();

    MarksResponse getMarksById(Long id);

    MarksResponse updateMarks(Long id, MarksRequest request);

    void deleteMarks(Long id);

    List<MarksResponse> getMyMarks();

    void saveMarks(List<MarksRequest> requests);

    List<MarksResponse> getMarksByCourse(Long courseId);
}